_source/_posts/2021-12-16-cloud-native-java-microservices-with-istio.md

```diff
@@ -0,0 +1,580 @@
+}
+
+microservice Notification with notification
+```
+
+We define entities for each service and mark the entities as microservice entities. We also define relationships between entities, enums and other options like pagination, service layer and so on. Please refer to JDL [Entities](https://www.jhipster.tech/jdl/entities-fields) and [relationships](https://www.jhipster.tech/jdl/relationships) documentation for more possibilities.
+
+Now, we are ready to run JHipster. Open a terminal window on the folder where you saved the JDL and run the below command.
+
+```bash
+$ jhipster jdl app.jdl --fork
+```
+
+This will create the applications with all its entities and specified configurations. You should be able to see the gateway application in action by running the below command on the **store** folder. You will be redirected to Keycloak for login.
+
+```bash
+$ docker-compose -f src/main/docker/keycloak.yml up -d # starts keycloak in daemon mode
+$ ./gradlew # starts the spring boot application
+```
+
+You can find a [sample application on GitHub](https://github.com/oktadev/okta-java-spring-k8s-istio-microservices-example).
+
+## Configure Okta as Identity Provider
+
+The applications are configured by default to use Keycloak as the identity provider since we choose OAuth2 as authentication mechanism. However, you can switch to any other OAuth2 provider by editing the **application.yml** file for each of the applications.
+
+Let's see how to configure Okta as the identity provider. You may wonder? Why Okta? Keycloak is great for development and testing but its not robust for production as you would need to manage it and keep it updated and so on. Okta on the other hand lets you worry about your business needs by keeping your identity requirements always on and secure.
+
+### Create an Okta application
+
+{% include setup/cli.md type="jhipster" %}
+
+### Configure OpenID Connect settings to use Okta
+
+Open `src/main/resources/config/application.yml` file for each application and update `issuer-uri`, `client-id` and `client-secret` under the `spring.security.oauth2.client` section as shown below with values you obtained during previous step.
+
+```yml
+spring:
+  ...
+  security:
+    oauth2:
+      client:
+        provider:
+          oidc:
+            issuer-uri: https://{yourOktaDomain}/oauth2/default
+        registration:
+          oidc:
+            client-id: {yourClientId}
+            client-secret:
+            scope: openid,profile,email
+```
+
+> **Note**: For security reasons we will not set the client-secret on this file and instead use environment variable, `SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET` or Kubernetes secrets. You can also skip updating the application.yml files altogether and use environment variable or Kubernetes secrets for all the OIDC configurations.
+
+You can run `./gradlew` on the store application to test the Okta login. Make sure to set the client secret environment variable first.
```

Comment:
I'm able to log in to the store using Okta, but I'm not able to navigate to any entities. I get 404 errors. Is that expected? If so, you should explain why.